src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -923,10 +931,28 @@ bool is_array_superset(
     // in draft 2020, "prefixItems" is used to represent tuples instead of an
     // overloaded "items"
     constexpr static auto get_tuple_items_kw = [](json_schema_dialect d) {
-        if (d == json_schema_dialect::draft202012) {
+        using enum json_schema_dialect;
+        switch (d) {
+        case draft4:
+        case draft6:
+        case draft7:
+        case draft201909:
+            return "items";
+        case draft202012:
             return "prefixItems";
         }
-        return "items";
+    };
+    constexpr static auto get_additional_items_kw = [](json_schema_dialect d) {
+        using enum json_schema_dialect;
+        switch (d) {
+        case draft4:
+        case draft6:
+        case draft7:
+        case draft201909:
+            return "additionalItems";
+        case draft202012:
+            return "items";
+        }
     };
```

Comment:
Note for the reviewers: I didn't factor out these functions because I think this keyword evolution doesn't make sense outside the context of is_array_superset.
I think the proper solution would be a normalization step to an internal representation that could abstract over these differences between dialects.

Reply:
> that would have to be bidirectional, right? 
yes but we store the schema dialect so we would be able to derive the correct kw to use, and in practice i think it would be easier to store a copy of the original json, because it could contain keys that we ignore for compat checks but that are meaningful for the user